arch/RISCV/RISCVInstPrinter.c

```diff
@@ -0,0 +1,355 @@
+/* By Nguyen Anh Quynh <aquynh@gmail.com>, 2013-2022, */
+/*    Rot127 <unisono@quyllur.org> 2022-2023 */
+/* Automatically translated source file from LLVM. */
+
+/* LLVM-commit: <commit> */
+/* LLVM-tag: <tag> */
+
+/* Only small edits allowed. */
+/* For multiple similar edits, please create a Patch for the translator. */
+
+/* Capstone's C++ file translator: */
+/* https://github.com/capstone-engine/capstone/tree/next/suite/auto-sync */
+
+//===-- RISCVInstPrinter.cpp - Convert RISC-V MCInst to asm syntax --------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This class prints an RISC-V MCInst to a .s file.
+//
+//===----------------------------------------------------------------------===//
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <capstone/platform.h>
+
+#include "RISCVMapping.h"
+#include "RISCVInstPrinter.h"
+
+#define GET_SUBTARGETINFO_ENUM
+#include "RISCVGenSubtargetInfo.inc"
+
+#define GET_INSTRINFO_ENUM
+#include "RISCVGenInstrInfo.inc"
+
+#define GET_REGINFO_ENUM
+#include "RISCVGenRegisterInfo.inc"
+
+#define CONCAT(a, b) CONCAT_(a, b)
+#define CONCAT_(a, b) a##_##b
+
+#define DEBUG_TYPE "asm-printer"
+
+// Include the auto-generated portion of the assembly writer.
+#define PRINT_ALIAS_INSTR
+#include "RISCVGenAsmWriter.inc"
+
+// Print architectural register names rather than the ABI names (such as x2
+// instead of sp).
+// TODO: Make RISCVInstPrinter::getRegisterName non-static so that this can a
+// member.
+static bool ArchRegNames;
+
+void printRegName(SStream *O, MCRegister Reg)
+{
+	SStream_concat0(markup(O, Markup_Register), getRegisterName(Reg));
+}
+
+static inline void printOperand(MCInst *MI, unsigned OpNo, SStream *O,
+				const char *Modifier)
+{
+	RISCV_add_cs_detail_0(MI, RISCV_OP_GROUP_Operand, OpNo);
+	CS_ASSERT((Modifier == nullptr || Modifier[0] == 0) &&
+		  "No modifiers supported");
+	MCOperand *MO = MCInst_getOperand(MI, (OpNo));
+
+	if (MCOperand_isReg(MO)) {
+		printRegName(O, MCOperand_getReg(MO));
+		return;
+	}
+
+	if (MCOperand_isImm(MO)) {
+		SStream_concat0(markup(O, Markup_Immediate),
+				formatImm(MCOperand_getImm(MO)));
+		return;
+	}
+
+	CS_ASSERT(MCOperand_isExpr(MO) &&
+		  "Unknown operand kind in printOperand");
+	MCOperand_getExpr(MO)->print(O, &MAI);
+}
+
+static inline void printBranchOperand(MCInst *MI, uint64_t Address,
+				      unsigned OpNo, SStream *O)
+{
+	RISCV_add_cs_detail_0(MI, RISCV_OP_GROUP_BranchOperand, OpNo);
+	MCOperand *MO = MCInst_getOperand(MI, (OpNo));
+	if (!MCOperand_isImm(MO))
+		return printOperand(MI, OpNo, O);
+
+	if (PrintBranchImmAsAddress) {
+		uint64_t Target = Address + MCOperand_getImm(MO);
+		if (!STI.hasFeature(RISCV_Feature64Bit))
```

Comment:
The `STI.hasFeature` should probably be automated away as well.
You can check the `STIFeatureBits` patch and just copy and adapt it if you have some time.